mantidproject/mantid
# Make background fit optional in CreateFloodWorkspace

**Description of work.**
Added properties and code to scale the flood to a given spectrum instead of a fitted line. The user has an option to choose between the two scaling methods.

**To test:**

Run the script:
```
ws1 = CreateFloodWorkspace(Filename='OFFSPEC0035946', CentralPixelSpectrum=400)
ws2 = CreateFloodWorkspace(Filename='OFFSPEC0035946', CentralPixelSpectrum=400, StartSpectrum=250, EndSpectrum=600)
ws3 = CreateFloodWorkspace(Filename='OFFSPEC0035946', CentralPixelSpectrum=400, ExcludeSpectra=[260, 261, 262, 516, 517, 518])
ws4 = CreateFloodWorkspace(Filename='OFFSPEC0035946', CentralPixelSpectrum=400, StartSpectrum=250, EndSpectrum=600, ExcludeSpectra=[260, 261, 262, 516, 517, 518])
```
Check that all output workspaces contain 1.0 in their spectrum 400. Check also that spectra outside the range of [StartSpectrum, EndSpectrum] have value 1.0 and the spectra from ExcludeSpectra have a very big number (1e200).

Fixes #23391 

*This does not require release notes because it's a correction to a newly created algorithm which has already been mentioned there.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
